shared/js/moz_intl.js

```diff
@@ -1,196 +1,520 @@
-(function(global) {
-'use strict';
-
-global.mozIntl = {
-  /**
-   * Format an Array of strings using locale specific separators.
-   * Currently it only uses middle separator resulting in a local
-   * equivalent of 'X, Y, Z'.
-   *
-   * In the Intl API it will support start/end separators allowing for
-   * things like 'X, Y and Z'.
-   *
-   * @param {Array} An array of strings to be formatted
-   * @returns {Promise} A promise of a string
-   */
-  formatList: function(list) {
-    return document.l10n.formatValue('listSeparator_middle').then(
-      sep => list.join(sep)
-    );
-  },
-
-  /**
-   * Return locale specific infromation about calendar system.
-   *
-   * Currently supports:
-   *   * firstDayOfTheWeek: 0 - Sunday, 1 - Monday, etc.
-   *
-   * @param {String} Identifier of a token to be retrieved
-   * @returns {Promise} A promise of value corresponding to a token
-   */
-  calendarInfo: function(token) {
+/* jshint ignore:start */
+
+var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();
```

Comment:
Unfortunately no. We currently rely on B2G 39 for build system and it uses Reflect API to build AST for JS and it doesnt support "class A extends B{}" syntax. Once we switch fully to node and use esprima (which does support es6), we'll be able to stop bablifying.